### PR TITLE
Reel board custom config pyocd

### LINKED ISSUE #32966: frdm_k64f: Run some testcases timeout failed by using twister
Describe the bug:
I find some testcases timeout failed after twister changed. More detail as below:

**frdm_k64f platform:**
Just select a testcase to demo this failure.
command: twister -p frdm_k64f --device-testing --device-serial /dev/ttyACM0 -T tests/kernel/stack/ -j 1 -vv 
and it showed me timeout failed, and device.log shows:
```
WARNING: runners.pyocd: hex file (None) does not exist; falling back on .bin (/home/zn/zephyrproject/zephyr/twister-out/frdm_k64f/tests/kernel/stack/stack/kernel.stack.usage/zephyr/zephyr.bin). Consider enabling CONFIG_BUILD_OUTPUT_HEX.
0005001:INFO:loader:Erased 77824 bytes (19 sectors), programmed 77824 bytes (19 pages), skipped 4096 bytes (1 page) at 19.53 kB/s
```
**And I reset to commit c5caa9b916d1 ,  It is good to work.**

**sam_e70b_xplained platform:**
command: twister -p sam_e70b_xplained --device-testing --device-serial /dev/ttyACM1 -T tests/kernel/stack/stack/  -j 1 -vv
and it showed me timeout failed, device.log shows:
```
Open On-Chip Debugger 0.10.0+dev-01508-gf79c90268-dirty (2021-01-24-19:03)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Info : flash bank command
Info : CMSIS-DAP: SWD  Supported
Info : CMSIS-DAP: FW Version = 03.25.01B6
Info : CMSIS-DAP: Serial# = ATML2637010000000000
Info : CMSIS-DAP: Interface Initialised (SWD)
Info : SWCLK/TCK = 1 SWDIO/TMS = 1 TDI = 1 TDO = 1 nTRST = 0 nRESET = 1
Info : CMSIS-DAP: Interface ready
Info : clock speed 1800 kHz
Info : SWD DPIDR 0x0bd11477
Info : atsame70q21.cpu: hardware has 8 breakpoints, 4 watchpoints
Info : starting gdb server for atsame70q21.cpu on 3333
Info : Listening on port 3333 for gdb connections
    TargetName         Type       Endian TapName            State       
--  ------------------ ---------- ------ ------------------ ------------
 0* atsame70q21.cpu    cortex_m   little atsame70q21.cpu    running

target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x00403638 msp: 0x20402c00
Info : device id = 0xa1020e01
Info : erasing lock regions 0-4...
Info : erasing lock region 0
Info : erasing lock region 1
Info : erasing lock region 2
Info : erasing lock region 3
Info : erasing lock region 4
auto erase enabled
wrote 81920 bytes from file /home/xuhao/work/zephyrproject/zephyr/twister-out/sam_e70b_xplained/tests/kernel/stack/stack/kernel.stack.usage/zephyr/zephyr.hex in 1.864619s (42.904 KiB/s)

target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x00402ec0 msp: 0x20402400
verified 70972 bytes in 1.719761s (40.301 KiB/s)

shutdown command invoked
```
**And I reset to commit c5caa9b916d1 ,  It is good to work.**

**mimxrt1050_evk platform:**
command: twister -p mimxrt1050_evk --device-testing --device-serial /dev/ttyACM0 -T tests/kernel/stack/stack/  -j 1 -vv
and it showed me build failed, and there is nothing wrong in build.log file.
**And I reset to commit c5caa9b916d1 ,  It is good to work.**

And reel_board platform is also not work, more detail refer above.

Environment
OS: Fedora33
Toolchain:zephyr-sdk-0.12.2
Commit ID: 4118ed1d4d6fe50f4

### PR DESCRIPTION
**Pull request by ioannisg**
Fixes #32966 (the fix is for the reel_board). 
The PR
- adds support in pyocd runner to attach a board-specific configuration file
- use a board-specific pyocd.yaml file for reel_board, to pass configuration for the flash command

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]
Nice, I didn't know about the pycod --config option.

@flit fyi

**Comment by flit** [comment]
Looks good.

One consideration is that pyocd currently only supports one config file (multiple are on the todo list), so having a board-specific config file will preclude a user/project config file.

File: boards/arm/reel_board/support/pyocd.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# reel_board global options
+
+# Use HW reset option on reel_board
+reset_type: 'hw'
+# Hold HW reset asserted for 0.5s.
```
Comment (ioannisg) [inline boards/arm/reel_board/support/pyocd.yaml:5]: comment needs update (todo)

**Comment by ioannisg** [comment]
> Looks good.
> 
> One consideration is that pyocd currently only supports one config file (multiple are on the todo list), so having a board-specific config file will preclude a user/project config file.

Is this something we could fix on the runner code? E.g. add logic so if both config files exist, pick the one that has precedence?

**Comment by flit** [comment]
If this is something you're concerned about, it might be better to just implement multiple config file support in pyocd. (Of course I would say that! 😉) The hardest part would be handling `--no-config` mixed into the command line.

For now, this PR is probably fine as is. If needed, the user can use `--tool-opt="--config=<path>"` to override the config file path.

**Review by nashif — APPROVED** [review]

**Pull request merged**